proposed/dotnet-bootstrapper-muxer-path.md

```diff
@@ -0,0 +1,68 @@
+
+In theory, this is 'doable' for .NET 10 in terms of changing the muxer to know about a specified bootstrapper location, which would enable these scenarios outside of just .NET 11.
+
+
+### Create a C# Muxer
+
+The muxer would become a bootstrapper and the muxer itself. It would cease to become C++ and become ported C# code that is run as native AOT code. This is high-risk and also likely still lower performance than a C++ application.
+
+
+## PATH Issue
+
+Visual Studio and the MSI or 'global' .NET Installer set the System `PATH` on Windows and not the user level `PATH`. The system `PATH` always takes precedence over the user `PATH`. If `dnvm` sets the user path to enable a folder of local .NET SDKs to be discovered by the `dotnet.exe` application host at `DOTNET_ROOT`, via the `PATH`, those installs would become 'undetectable' to the runtime, debugger, etc when Visual Studio would update and modify the System `PATH`.
+
+This is also applicable to `source-built` SDKs and the `pkg` installer.
+
+There have been several methods suggested to handle this. None of them are ideal.
+
+### Fail on Windows if System PATH set, if `paths` in `global.json` is not set
+
+In this approach, we prioritized defined and replicable behavior over the easier UX behavior. System administrator installs are created in predictable locations, especially those from Visual Studio and/or package manager feeds.
+
+What we could do is look for those feeds per install operation and decline to do anything when a user tries to run `dotnet install`. The error message would point the user towards leveraging the `sdk: paths []` option in their repository's `global.json` file, which would specify to the host or `dotnet.exe` where to look for the installs of the .NET SDK or .NET runtimes. The host is already aware of this as of .NET 10.
+
+This is subject to backlash, especially as only 2 to 6% of users are on a `global.json` according to internal metrics.
+
+However, there is precedence in the sense that `nvm` is a Linux and macOS only tool, from an official standpoint. An alternative would be to make `dotnet install` only applicable to Linux and macOS. In this case, I would be more strongly opposed to using `dotnet` as the root command, and instead leaning towards making this an 'opinionated' installer that resolves the Linux and macOS scenario. This creates a split between OS support.
+
+### Update the Installers / VS to Set the User PATH
+
+The Windows official guidance is to set the system level PATH, not to set the global `PATH`. However, there are some possible issues with setting the system `PATH`. For one, we believe that the Windows API for setting the user path does a `COM` broadcast to every single open window on the OS and waits for a response. `chrome.exe`, for example, hangs for 1 minute when receiving this broadcast, which means setting the API for setting the system PATH can take a minute, mostly dead waiting.
```

Comment:
I'm a bit confused by the terminology here. I think of 'system' and 'global' as the same. Where we currently set 'system', but are considering setting 'user'.

Reply:
Great point, this wording is confusing. I rewrote this portion and some other parts. Some of this was incorrect.